### DISCUSSION TITLE
[Headless CMS] Export table with lookup on auth0 users

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#298] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11

### Description

Hi guys,
I have the following situation on a client's namespace.
* cms-site v9.14.2
* cms-backend v5.1.2
* v1-adapter v4.1.3
* export-service v1.0.6
* crud-service v6.3.0

The client still has the old headless CMS. The CMS access is handled through Auth0 and, in the CMS, there is a "users" page that shows users from Auth0 (with advanced configurations, you know).

The client has an `items` collection with a `lastActionUserId` string property that contains the id of the user on Auth0.
The `items` collection is used on the CMS for an `items` page that has the `lastActionUserId` configured to lookup from the `users` collection, containing the Auth0 users data:
```json
{
    "lastActionUserId": {
        "id": "lastActionUserId",
        "type": "string",
        "required": false,
        "label": "Last    Action User",
        "interfaceType": "cmslookup",
        "cmsLookup": {
            "text": {
                "sources": [
                    "username"
                ],
                "delimiters": []
            },
            "value": "id",
            "autoSelect": false,
            "reset": false,
            "limit": 20,
            "pageSize": 100,
            "query": "",
            "searchLive": false,
            "collectionIdSource": "users"
        },
        "description": "",
        "cmsOrder": 570,
        "cmsCardPosition": 0,
        "cmsEditable": false,
        "cmsVisibility": {
            "level": 2
        }
    }
}
```

The problem is that the export with lookup resolution fails.
I tried two different approaches.

# Test 1 - No "users" collection in "MongoDB CRUD" console section
Without the `users` collection in the "MongoDB CRUD" section, with the associated endpoint, the query produced by the v1-adapter for the lookup is wrong, because the `$lookup` operator of the `lastActionUserId` is wrong, the `from` property is missing, as reported into the attached log `v1-adapter-wrong-lookup.log`.
[v1-adapter-wrong-lookup.log](https://github.com/mia-platform/community/files/12214341/v1-adapter-wrong-lookup.log)

# Test 2 - "users" collection added and exposed from the console
I tried to add and expose the `users` collection on the console and, after a deploy (and a manual edit of the `v1-adapter` configmap), the export query was right, but I got another error, as reported into the attached log `v1-adapter-right-lookup.log`.
[v1-adapter-right-lookup.log](https://github.com/mia-platform/community/files/12214345/v1-adapter-right-lookup.log)

Can someone help me? Do you ever have a similar situation?

Thanks in advance, chat me for missing details.

p.s. the export with a lookup on other CRUD (not the `users` one) works well.

### Actual Outcome

The export fails.

### Expected Outcome

The export works.

**Comment by davidebianchi** [comment, marked as answer]
Hi @silversoul93
We update for a similar issue the v1-adapter to v4.1.4, but we don't upgrade it with Console since it is deprecated. Can you try it with this new version to see if it could fix the issue?

**Reply by frengooo** [reply]
Hi @davidebianchi, thanks for your reply.
With the v4.1.4 I got the first error again ("Test 1 ..."), even if the `users` collection is present and exposed.

**Reply by davidebianchi** [reply]
Okay, I think that the problem is that it's missing the users collection, so it should be enabled [the sync user feature of auth0-client](https://docs.mia-platform.eu/docs/runtime_suite/auth0-client/configuration#environment-variables).
If users it is exposed, it is quite strange that it results in the missing from error 🤔
